src/ui/Header/HeaderMenuPopUp.js

```diff
@@ -33,102 +57,147 @@ const ItemGrid = ({ sectionTitle, item, columns, renderMenuItem }) => {
+    layout.hideChildrenFromNavigation === undefined
+      ? true
+      : layout.hideChildrenFromNavigation;
+  return !layout.childrenColumns || layout.childrenColumns[index] === 1 ? (
+    <Item
+      item={section}
+      renderMenuItem={renderMenuItem}
+      hideChildrenFromNavigation={hideChildrenFromNavigation}
+    />
+  ) : (
+    <ItemGrid
+      sectionTitle={section.title}
+      item={section}
+      columns={layout.childrenColumns[index]}
+      renderMenuItem={renderMenuItem}
+      hideChildrenFromNavigation={hideChildrenFromNavigation}
+    />
+  );
+};
 
-const Countries = ({ menuItem, renderMenuItem }) => (
-  <Grid>
-    <Grid.Column width={8}>
-      {menuItem.items.map((section, index) => (
-        <React.Fragment key={index}>
-          {index === 0 && (
-            <ItemGrid
-              sectionTitle={section.title}
-              item={section}
-              columns={5}
-              renderMenuItem={renderMenuItem}
-            />
-          )}
-        </React.Fragment>
-      ))}
-    </Grid.Column>
-    <Grid.Column width={4}>
-      <Grid columns={1} className="nested-grid">
-        {menuItem.items.map((section, index) => (
+const StandardMegaMenuGrid = ({ menuItem, renderMenuItem, layout }) => {
+  return layout && layout.columnsWidth ? (
+    <Grid>
+      {layout.itemsEquallySpread &&
+        menuItem.items.map((section, index) => (
           <React.Fragment key={index}>
-            {index !== 0 && (
-              <Grid.Column>
-                <ItemGrid
-                  sectionTitle={section.title}
-                  item={section}
-                  columns={2}
-                  renderMenuItem={renderMenuItem}
-                />
-              </Grid.Column>
-            )}
+            <div className={layout.columnsWidth[index]}>
+              <RenderItem
+                layout={layout}
+                section={section}
+                renderMenuItem={renderMenuItem}
+                index={index}
+              />
+            </div>
           </React.Fragment>
         ))}
-      </Grid>
-    </Grid.Column>
-  </Grid>
-);
+      {!layout.itemsEquallySpread &&
+        layout.columnsWidth.map((columnWidth, index) => {
+          const columns = layout.columnsWidth.length;
+          const isLastColumn = columns - 1 === index;
+          const lastColumnItems =
+            isLastColumn && menuItem.items.length >= layout.columnsWidth.length
+              ? menuItem.items.slice(index)
+              : [];
 
-const StandardMegaMenuGrid = ({ menuItem, renderMenuItem }) => (
-  <Grid columns={4}>
-    {menuItem.items.map((section, index) => (
-      <Grid.Column key={index}>
-        <Item item={section} renderMenuItem={renderMenuItem} />
-      </Grid.Column>
-    ))}
-  </Grid>
-);
+          const itemsMatrix = listToMatrix(menuItem.items, columns);
+          return (
+            <div key={index} className={layout.columnsWidth[index]}>
+              {layout.itemsEquallySpread &&
```

Comment:
@MihaelaCretu11 this code will never run since the check at line 132 is for not having       {!layout.itemsEquallySpread &&